src/devicecontroller/DefaultDeviceController.ts

```diff
@@ -489,10 +562,17 @@ export default class DefaultDeviceController implements DeviceControllerBasedMed
     }
 
     for (const track of tracksToStop) {
-      this.logger.info(`stopping ${track.kind} track`);
       track.stop();
     }
 
+    // Calling `stop` does not actually trigger `onended` callbacks unless we do
+    // it manually. Do so in order to clean up listeners, if for no other reason.
+    AsyncScheduler.nextTick(() => {
+      for (const track of tracksToStop) {
+        track.dispatchEvent(new Event('ended'));
+      }
+    });
+
     for (const kind in this.activeDevices) {
       if (this.activeDevices[kind] && this.activeDevices[kind].stream === mediaStreamToRelease) {
         this.activeDevices[kind] = null;
```

Comment:
nit: can we delete this entry instead of setting to null

Reply:
Sure!